go/libraries/doltcore/mvdata/engine_table_writer.go

```diff
@@ -187,8 +187,15 @@ func (s *SqlEngineTableWriter) WriteRows(ctx context.Context, inputChannel chan
 		}
 	}()
 
-	line := 1
+	// If there were create table statements, they are automatically committed, so we need to start a new transaction
+	if s.importOption == CreateOp {
+		_, _, _, err = s.se.Query(s.sqlCtx, "START TRANSACTION")
```

Comment:
Always drain the row iter for these things, we often don't throw any error until iteration or even Close()